routers/repo/http.go

```diff
@@ -95,6 +95,13 @@ func HTTP(ctx *context.Context) {
 	}
 
 	repo, err := models.GetRepositoryByName(repoUser.ID, reponame)
+	if models.IsErrRepoNotExist(err) && setting.Repository.Upload.AutoCreate {
+
+		// if 'CreateRepository' fails, the error are handled in the next 'if err != nil' block
+		repo, err = models.CreateRepository(repoUser, repoUser, models.CreateRepoOptions{
```

Comment:
What about access rights? Logged user don't need to be same as `repoUser`. You need user is logged and if he can create repository under `repoUser` (e.g. is admin or wants to add repository to his organization or organization he is in owner team).